docs/developer/release/prepare.md

```diff
@@ -0,0 +1,42 @@
+# Prepare build environments
```

Comment:
Maybe name this `{setup,prepare,update}-build-environments`? It's a bit long but just `prepare` doesn't cut it for me as its a bit too broad (what are we preparing? It could be misunderstood for the pre-release steps.)

Reply:
I originally had it as `prepare-build-environments.md`, so I'll switch back to that.